### PR TITLE
Implement system upgrade for in-place re-partitioning

### PR DESCRIPTION
**Pull request by carlroth**
Reviewer: @jnealtowns @sonoble 

Note that this will not work for self-contained (SWI-included) installers.

### CODE REVIEW

**Comment by carlroth** [comment]
Sorry, my patch

WIP parted API updates -- DO NO MERGE

made it in with the wrong commit message. I cleaned it up later in the PR but neglected to change the wording. The contents of the commit are valid.

**Comment by carlroth** [comment]
TBD update upgrade-shell et al to include the proper fixups for /tmp, /etc/onl

**Comment by carlroth** [comment]
ping @jnealtowns @sonoble i'm back in town now, can we move on this?

**Comment by jnealtowns** [comment]
@carlroth  Yes I will start the review.

File: packages/base/all/vendor-config-onl/src/python/onl/install/SystemInstall.py [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+    def __init__(self, callback=None, log=None):
+        super(UpgradeHelper, self).__init__(log=log)
+        self.callback = callback
+
+    def _runInitrdShell(self, p):
+        if self.callback is not None:
+            self.callback(self, p)
+
+class OnlBootContext(object):
```
Comment (jnealtowns) [inline packages/base/all/vendor-config-onl/src/python/onl/install/SystemInstall.py]: You should use the existing OnlMountReadWriteContext("ONL-BOOT") from onl.mounts

File: packages/base/all/vendor-config-onl/src/python/onl/install/SystemInstall.py [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+            except AttributeError:
+                l = []
+            for f in l:
+                src = os.path.join(sdir, f)
+                if os.path.exists(src):
+                    dst = os.path.join(abs_idir, f)
+                    self.log.debug("+ /bin/cp %s %s", src, dst)
+                    shutil.copy2(src, dst)
+
+            with OnlBootContext(log=self.log) as octx:
```
Comment (jnealtowns) [inline packages/base/all/vendor-config-onl/src/python/onl/install/SystemInstall.py]: with OnlMountReadWriteContext("ONL-BOOT", logger=self.log):

File: packages/base/all/vendor-config-onl/src/python/onl/install/SystemInstall.py [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+
+        h, self.onieHelper = self.onieHelper, None
+        if h is not None:
+            h.shutdown()
+
+    @classmethod
+    def main(cls):
+
+        logging.basicConfig()
+        logger = logging.getLogger("swl-install")
```
Comment (jnealtowns) [inline packages/base/all/vendor-config-onl/src/python/onl/install/SystemInstall.py]: onl-install

File: packages/base/all/vendor-config-onl/src/python/onl/install/SystemInstall.py (hunk 1 above)
Comment (carlroth) [inline packages/base/all/vendor-config-onl/src/python/onl/install/SystemInstall.py]: I'll look into that. The code I'm using will also re-use an existing mount of a devide or label, regardless of mount-point.

Comment (jnealtowns) [inline packages/base/all/vendor-config-onl/src/python/onl/install/SystemInstall.py]: So does the OnlMountContext.

**Pull request merged**